snowexsql/api.py

```diff
@@ -270,9 +314,88 @@ class LayerMeasurements(PointMeasurements):
     ]
     # TODO: layer analysis methods?
 
+    @property
+    def all_site_ids(self):
+        """
+        Return all types of the data
+        """
+        with db_session(self.DB_NAME) as (session, engine):
+            qry = session.query(self.MODEL.site_id).distinct()
+            result = qry.all()
+        return self.retrieve_single_value_result(result)
 
 class RasterMeasurements(BaseDataset):
     MODEL = ImageData
+    ALLOWED_QRY_KWARGS = BaseDataset.ALLOWED_QRY_KWARGS + ['description']
+
+    @property
+    def all_descriptions(self):
+        with db_session(self.DB_NAME) as (session, engine):
+            qry = session.query(self.MODEL.description).distinct()
+            result = qry.all()
+        return self.retrieve_single_value_result(result)
+
+    @classmethod
+    def check_for_single_dataset(cls, **kwargs):
+        """
+        At the moment there is not a clear path to how to deal with multiple rasters so
+        check that the user only requested one dataset
+        """
+        LOG.info("Checking raster query for single raster dataset...")
+        multi_raster_indicators = ['instrument', 'date', 'observers', 'doi', 'type', 'description']
+        with db_session(cls.DB_NAME) as (session, engine):
+            try:
+                # Form query and check if the query spans multipl rasters
+                for column in multi_raster_indicators:
+                    values = cls.from_unique_entries([column], **kwargs)
+                    if len(values) > 1:
+                        options = [f"'{v}'" for v in values]
+                        raise TooManyRastersException(f"More than one `{column}` suggests there are multiple raster datasets. "
+                                                      f"Try filter {column} to one of the following values {', '.join(options)}.")
+
+            except Exception as e:
+                session.close()
```

Comment:
I don't think you need to call session lose. I believe the `db_session` should handle that